### PR TITLE
Global Styles: Use block settings on the block panels.

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
Currently, on the pannels of each block, we don't use the settings for that block.
So for example, if the paragraph has a specific color palette when we go to the paragraph settings on GloboStyles as options for text and background color we don't see that palette and we just see the global one.

This PR fixes that and makes sure the settings used are the ones of the block whose styles we are editing.

## How has this been tested?
Open the global styles sidebar.
Go to the paragraph block settings.
Add a block specific color palette for the paragraph.
Verify the background and text color options show the new user-defined palette instead of the global one.

### CODE REVIEW

**Comment by nosolosw** [comment]
:wave: I've taken a quick look at this and these are the main points I see:

- The logic is the same for every setting (use your own if it exists, otherwise fall back to the global value) & the values provided by `getMergedSetting` are the same present in the `core/edit-site` store. What if we created a `useEditorFeature` hook for the edit-site that centralizes all of this logic and takes the values from the store instead, as we did for block editor store?

- This PR enables the block-editor/component/font-size-picker to take more properties, which it was supposed to abstract from the consumer (see readme). I was wondering if we should either 1) use the base component directly here or 2) create a font size picker component for edit-site that uses the hook I mentioned above?

**Comment by jorgefilipecosta** [comment]
Hi @nosolosw, thank you for the review! Your feedback was applied 👍

File: packages/edit-site/src/components/editor/utils.js [hunk 1]
```diff
@@ -20,3 +29,15 @@ export const fromPx = ( value ) => {
...
+
+export function useEditorFeature( featurePath, blockName = GLOBAL_CONTEXT ) {
+	const settings = useSelect( ( select ) => {
+		return select( 'core/edit-site' ).getSettings();
+	} );
+	return (
+		get(
+			settings,
+			`__experimentalFeatures.${ blockName }.${ featurePath }`
+		) ?? get( settings, `__experimentalFeatures.global.${ featurePath }` )
```
Comment (nosolosw) [inline packages/edit-site/src/components/editor/utils.js:41]:
```suggestion
		) ?? get( settings, `__experimentalFeatures.${ GLOBAL_CONTEXT }.${ featurePath }` )
```

**Review by oandregal — APPROVED** [review]
Left a minor comment, but this is a very nice PR now! :muscle:

**Pull request merged**